tests/read_test.h

```diff
@@ -0,0 +1,226 @@
+#ifndef PODIO_TESTS_READ_TEST_H_
+#define PODIO_TESTS_READ_TEST_H_
+// test data model
+#include "datamodel/ExampleHitCollection.h"
+#include "datamodel/ExampleClusterCollection.h"
+#include "datamodel/ExampleMCCollection.h"
+#include "datamodel/ExampleReferencingTypeCollection.h"
+#include "datamodel/ExampleWithOneRelationCollection.h"
+#include "datamodel/ExampleWithVectorMemberCollection.h"
+#include "datamodel/ExampleWithComponentCollection.h"
+#include "datamodel/ExampleWithARelationCollection.h"
+#include "datamodel/ExampleWithStringCollection.h"
+#include "datamodel/ExampleWithNamespace.h"
+#include "datamodel/ExampleWithArrayCollection.h"
+
+// podio specific includes
+#include "podio/EventStore.h"
+
+// STL
+#include <vector>
+#include <iostream>
+#include <exception>
+#include <cassert>
+#include <sstream>
+
+
+void processEvent(podio::EventStore& store, bool verboser, unsigned eventNum) {
+
+  auto evtMD = store.getEventMetaData() ;
+  float evtWeight = evtMD.getFloatVal( "UserEventWeight" ) ;
+  if( evtWeight != (float) 100.*eventNum ){
+    std::cout << " read UserEventWeight: " << evtWeight << " - expected : " << (float) 100.*eventNum << std::endl ;
+    throw std::runtime_error("Couldn't read event meta data parameters 'UserEventWeight'");
+  }
+  std::stringstream ss ; ss << " event_number_" << eventNum ;
+  std::string evtName = evtMD.getStringVal( "UserEventName" ) ;
```

Comment:
If you would have used *const auto*, you would have gotten *const std::string&* instead of a copy.

Reply:
Ah yes, this is true. But these changes are currently only here because this PR starts on top of #133. I will fix it there. There are probably also a few other instances, where the usage of `auto` would make things a bit nicer in the tests. I haven't really checked, as I have basically just moved the already available code into a common header.